components/Promotion/interfaces.rst

```diff
@@ -0,0 +1,199 @@
+the ``PromotionCouponAwareSubjectInterface``.
+
+.. note::
+
+    This interface extends the :ref:`component_promotion_model_promotion-subject-interface`. |br|
+    You will find more information about this interface in `Sylius API PromotionCouponAwareSubjectInterface`_.
+
+.. _Sylius API PromotionCouponAwareSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionCouponAwareSubjectInterface.html
+
+
+.. _component_promotion_model_promotion-coupons-aware-subject-interface:
+
+PromotionCouponsAwareSubjectInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To make the object able to get its associated coupons collection, the object class needs to implement
+the ``PromotionCouponsAwareSubjectInterface``.
+
+.. note::
+
+    This interface extends the :ref:`component_promotion_model_promotion-subject-interface`. |br|
+    You will find more information about this interface in `Sylius API PromotionCouponsAwareSubjectInterface`_.
+
+.. _Sylius API PromotionCouponsAwareSubjectInterface: http://api.sylius.org/Sylius/Component/Promotion/Model/PromotionCouponsAwareSubjectInterface.html
+
+
+Services Interfaces
+-------------------
+
+.. _component_promotion_model_action-action-interface:
+
+PromotionActionInterface
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by models that do actions on promotions. These are the models that execute actions on the promotion subjects.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API PromotionActionInterface`_.
+
+.. _Sylius API PromotionActionInterface: http://api.sylius.org/Sylius/Component/Promotion/Action/PromotionActionInterface.html
+
+
+.. _component_promotion_checker_promotion-eligibility-checker-interface:
+
+PromotionEligibilityCheckerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to check the promotions' eligibility, the object class has to implement the ``PromotionEligibilityCheckerInterface``.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API PromotionEligibilityCheckerInterface`_.
+
+.. _Sylius API PromotionEligibilityCheckerInterface: http://api.sylius.org/Sylius/Component/Promotion/Checker/PromotionEligibilityCheckerInterface.html
+
+
+.. _component_promotion_checker_promotion-rule-checker-interface:
+
+RuleCheckerInterface
+~~~~~~~~~~~~~~~~~~~~
+
+In order to be able to check the eligibility of the rules, the object class needs to implement the ``RuleCheckerInterface``.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API RuleCheckerInterface`_.
+
+.. _Sylius API RuleCheckerInterface: http://api.sylius.org/Sylius/Component/Promotion/Checker/RuleCheckerInterface.html
+
+
+.. _component_promotion_action_promotion-applicator-interface:
+
+PromotionApplicatorInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To make the object able to apply promotions it needs to implement the ``PromotionApplicatorInterface``.
```

Comment:
Like I said, I think it sounds better to say that this should be implemented by promotion applying service. Usually this is just one service and this suggests you can do it all over the place, which is not recommended.